external/kafka/src/main/scala/org/apache/spark/streaming/kafka/ReliableKafkaReceiver.scala

```diff
@@ -78,10 +78,10 @@ class ReliableKafkaReceiver[
   private var blockOffsetMap: ConcurrentHashMap[StreamBlockId, Map[TopicAndPartition, Long]] = null
 
   /**
-   * Manage the BlockGenerator in receiver itself for better managing block store and offset
-   * commit.
+   * A BlockGenerator managed by the receiver itself for more control over storing blocks and
+   * committing offsets.
    */
-  private var blockGenerator: BlockGenerator = null
+  private lazy val blockGenerator = supervisor.createBlockGenerator(new GeneratedBlockHandler)
```

Comment:
The previous implementation will create a new `BlockGenerator` when restarting the receiver. Is it safe to reuse it?